### PR TITLE
fix_streetname_widget_shield

### PR DESCRIPTION
**Pull request by nnngrach**
[issue](https://github.com/osmandapp/OsmAnd-iOS/issues/2014)
[video](https://www.dropbox.com/s/fbjiw51ebf99zmy/Screen%20Recording%202022-07-11%20at%2011.49.22.mov?raw=1)

### CODE REVIEW

File: Sources/Controllers/Map/Widgets/OATopTextView.mm [hunk 1]
```diff
@@ -497,7 +497,7 @@ - (BOOL) updateInfo
         {
             if ([_settings.showStreetName get])
             {
-                OANextDirectionInfo *nextDirInfo = [_routingHelper getNextRouteDirectionInfo:_calc1 toSpeak:YES];
+                OANextDirectionInfo *nextDirInfo = [_routingHelper getNextRouteDirectionInfo:[[OANextDirectionInfo alloc] init] toSpeak:YES];
```
Comment (alex-osm) [inline Sources/Controllers/Map/Widgets/OATopTextView.mm:500]: Why _calc1 replaced to [[OANextDirectionInfo alloc] init]?

File: Sources/Router/OARouteCalculationResult.mm [hunk 2]
```diff
@@ -1176,19 +1337,20 @@ + (NSString *) toString:(std::shared_ptr<TurnType>)type shortName:(BOOL)shortNam
...
-                    CLLocation *prev = locations[locations.count - 1];
-                    lat = prev.coordinate.latitude;
-                    lon = prev.coordinate.longitude;
-                    if (prev.altitude)
-                        alt = @(prev.altitude);
-                }
-            }
-            [locations addObject:[[CLLocation alloc] initWithCoordinate:CLLocationCoordinate2DMake(lat, lon) altitude:alt? alt.doubleValue : NAN horizontalAccuracy:0 verticalAccuracy:0 timestamp:[NSDate date]]];
+//            // FIXME: investigate gpx file
+//            if (s->object->pointsX[i] == 0 && s->object->pointsY[i] == 0)
```
Comment (alex-osm) [inline Sources/Router/OARouteCalculationResult.mm:1341]: Why commented?

File: Sources/Router/OARouteProvider.mm [hunk 3]
```diff
@@ -648,7 +658,7 @@ - (OARouteCalculationResult *) emptyResult
     
     float mb = (1 << 20);
     natural_t freeMemory = [OAUtilities get_free_memory];
-    long memoryLimit = (0.1 * ([NSProcessInfo processInfo].physicalMemory / mb));
+    long memoryLimit = (0.95 * ([NSProcessInfo processInfo].physicalMemory / mb));
```
Comment (alex-osm) [inline Sources/Router/OARouteProvider.mm:661]: ?

File: Sources/Controllers/Map/Widgets/OATopTextView.mm (hunk 1 above)
Comment (nnngrach) [reply Sources/Controllers/Map/Widgets/OATopTextView.mm:500]: https://github.com/osmandapp/OsmAnd/blob/44259765c7829cf199f945d90d9583190bab5bd6/OsmAnd/src/net/osmand/plus/views/mapwidgets/widgets/StreetNameWidget.java#L427

**Pull request merged**